src/extensions/scratch3_vizblocks/index.js

```diff
@@ -136,6 +155,268 @@ class Scratch3VizBlocks {
+                    blockType: BlockType.COMMAND,
+                    text: formatMessage({
+                        id: 'vizblocks.clear',
+                        default: 'clear',
+                        description: 'clear canvas'
+                    })
+                }
+            ]
+        };
+    }
+
+    /**
+     * The pen "clear" block clears the pen layer's contents.
+     * @param {object} args - the block arguments.
+     * @param {object} util - utility object provided by the runtime.
+     */
+    clear (args, util) {
+        const penSkinId = this._getPenLayerID();
+        const target = util.target;
+        if (penSkinId >= 0) {
+            this.runtime.renderer.penClear(penSkinId);
+            this.runtime.requestRedraw();
+        }
+
+        if (!this._posEmpty){
+            this._xPos.length = 0;
+            this._yPos.length = 0;
+            this._posEmpty = true;
+        }
+
+        target.sprite.costumes_.forEach(costume => {
+            if (costume.name !== 'costume1' && costume.name !== 'costume2') {
+                // eslint-disable-next-line no-console
+                console.log(costume.name);
+                target.deleteCostume(target.getCostumeIndexByName(costume.name));
+            }
+        });
+
+    }
+
+    /**
+     * Draw X axis using functions from "pen block"
+     * @param {object} args - the block arguments.
+     * @param {object} util - utility object provided by the runtime.
+     */
+    drawAxisX (args, util) {
+        const target = util.target;
+        
+        // console.log('draw axis');
+        target.setVisible(false);
+        this._costumes = this.loadCostumes(util);
+
+        const penSkinId = this._getPenLayerID();
+        if (penSkinId >= 0) {
+            const penState = this._getPenState(target);
+            target.x = this._xCenter + this._width;
+            target.y = this._yCenter;
+            this.runtime.renderer.penLine(penSkinId, penState.penAttributes, this._xCenter, this._yCenter, target.x, target.y);
+
+            for (let i = 0; i * this._interval < this._width; i++){
+                const interval = i * this._interval;
+                this.runtime.renderer.penLine(penSkinId, penState.penAttributes, this._xCenter + interval, this._yCenter, this._xCenter + interval, this._yCenter + 5);
+            }
+
+            for (let i = 0; i < 10; i++){
+                this.setText(this._xCenter + (i * this._interval), this._yCenter - 10, i, util);
+            }
+            
+            this.runtime.requestRedraw();
+
+        }
+    }
+
+    /**
+     * Draw Y axis using functions from "pen block"
+     * @param {object} args - the block arguments.
+     * @param {object} util - utility object provided by the runtime.
+     */
+    drawAxisY (args, util) {
+        const target = util.target;
+        
+        // console.log('draw axis');
+        target.setVisible(false);
+        this._costumes = this.loadCostumes(util);
```

Comment:
We can pass `target` into the function directly.

```suggestion
        this._costumes = this.loadCostumes(target);
```